### PR TITLE
Add support for attribute alias

### PR DESCRIPTION
**Pull request by martndemus**
See #589

### CODE REVIEW

File: lib/ransack/adapters/active_record/base.rb [hunk 1]
```diff
@@ -23,9 +23,14 @@ def ransacker(name, opts = {}, &block)
...
         def ransackable_attributes(auth_object = nil)
-          column_names + _ransackers.keys
+          ransackable_attributes = column_names + _ransackers.keys
+
+          if Ransack::SUPPORTS_ATTRIBUTE_ALIAS
+            ransackable_attributes += attribute_aliases.keys
+          end
+
+          ransackable_attributes
         end
```
Comment (jonatack) [inline lib/ransack/adapters/active_record/base.rb]: We could probably just inline the values here without memoizing `column_names + _ransackers.keys` in a local var:

```ruby
if Ransack::SUPPORTS_ATTRIBUTE_ALIAS
  column_names + _ransackers.keys + attribute_aliases.keys
else
  column_names + _ransackers.keys
end
```

File: lib/ransack.rb [hunk 2]
```diff
@@ -8,6 +8,12 @@
 module Ransack
   extend Configuration
   class UntraversableAssociationError < StandardError; end;
+
+  SUPPORTS_ATTRIBUTE_ALIAS = begin
+    ActiveRecord::Base.respond_to?(:attribute_aliases)
+  rescue NameError
+    false
+  end
 end
```
Comment (jonatack) [inline lib/ransack.rb]: Do you know if there are versions of Active Record that do not support `attribute_aliases`?

**Comment by jonatack** [comment]
Thanks! This looks pretty good. Can you let me know for the question in the code?

File: lib/ransack.rb (hunk 2 above)
Comment (martndemus) [inline lib/ransack.rb]: `attribute_aliases` was introduced in Rails 4. So 3.x versions do not support it. I haven't found an alternative method in Rails 3.x that retrieves the aliases.

Comment (jonatack) [inline lib/ransack.rb]: Perfect, thanks.

**Comment by martndemus** [comment]
Inlined the things in the `ransackable_attributes` method.

**Comment by jonatack** [comment]
Thanks! :green_apple:  Merged [here](https://github.com/activerecord-hackery/ransack/commit/549342aeed2a5e9962c43f14f744930511d86de8).

**Comment by jonatack** [comment]
Would you like to make a PR to update the change log?

**Comment by martndemus** [comment]
@jonatack:  see #594

**Pull request closed without merging**